### PR TITLE
1000 breadcrumb title

### PR DESCRIPTION
**Pull request by shon-button**
[Card](https://github.com/bcgov/cas-registration/issues/1000)

## 🚀 Impact:
   - Replaces UUID in breadcrumbs with a more descriptive text 

##  🔬 Testing:

**Pre-reqs**
1. From terminal command, start the api server:
 ```
cd bc_obps
make reset_db
make run
```  
2.  From terminal command, start the app development server:
 ```
cd client && yarn build && yarn start
```  
3. From terminal command, add your IDIR user as `cas_admin` role through psql tool

```
cd bc_obps && psql
```

```
 \c registration
```

```
INSERT INTO erc.user (user_guid, app_role_id, first_name, last_name , position_title , email, phone_number, business_guid, bceid_business_name )
VALUES ('YOUR_USER_GUID_NO_HYPHENS','cas_admin','first_name','last_name ','position_title ','email@email.com','16044015477','efb76d57-88b7-4eb6-9f26-ec12b49c14c1','bceid_business_name')
ON CONFLICT (user_guid)
DO UPDATE SET app_role_id = EXCLUDED.app_role_id, first_name = EXCLUDED.first_name, last_name = EXCLUDED.last_name;
```

**Select Operator**
1. Navigate to `http://localhost:3000`
2. Click button `Log in with Business BCeID`
3. Log in with Bceid: `bc-cas-dev-three`
5. Complete and submit the Profile form
6. From the dashboard, click link `Select Operator\1 pending action(s) required`
7. Search and select an operator
8. Click button `Select Operator`
9. **Expected result**: selected operator's name is in breadcrumb
10. Continue the flow of selecting an operator
11. **Expected result**: selected operator's name remains in breadcrumb

**My Operator**
1. Navigate to `http://localhost:3000`
2. Click button `Log in with Business BCeID`
3. Log in with Bceid: `bc-cas-dev`
4. From the dashboard, click link `Select Operator\0 pending action(s) required`
5. **Expected result**: operator's name is in breadcrumb
6. Edit the operator
7. **Expected result**: selected operator's name remains in breadcrumb
8. Change the operator name and submit form
9. . From the dashboard, click link `Select Operator\0 pending action(s) required`
10. **Expected result**: operator's new name is in breadcrumb
12. From the dashboard, click link `My Operations`
13. From `Operations` grid click button "Create Operation"
14. Complete the flow of creating an operation including clicking the `Back` button
15. **Expected result**: available operation name is in breadcrumb
16. . From `Operations` grid click action column to view an operation
17.  **Expected result**: operation name is in breadcrumb
18. Navigate the operation pages
19. **Expected result**: selected operation's name remains in breadcrumb

**Operators**
1. Navigate to `http://localhost:3000`
2. Click button `Log in with IDIR`
3. Log in with your IDIR
4. From the dashboard, click link `Operators`
5. From `Operators` grid click action column to view an operator
6.  **Expected result**: operator's name is in breadcrumb
7.  Complete the Approve\Deny flow for an operator
8. **Expected result**: operator's name remains in breadcrumb

**Operations**
1. Navigate to `http://localhost:3000`
2. Click button `Log in with IDIR`
3. Log in with your IDIR
4. From the dashboard, click link `Operations`
5. From `Operations` grid click action column to view an operation
6.  **Expected result**: operation name is in breadcrumb
7. Navigate the operation pages
8. **Expected result**: selected operation's name remains in breadcrumb

### CODE REVIEW

File: client/app/components/navigation/Bread.tsx [hunk 1]
```diff
@@ -96,7 +87,26 @@ export default function Bread({
...
+  // and useState which is maintained between renders of a top-level React component (required for next\back) navigations
+  const searchParams = useSearchParams();
+  const paramTitle = searchParams.get("title") as string;
+  const [crumbTitle, setCrumbTitle] = useState<string>("");
+  useEffect(() => {
+    // Set the title state to rowTitle if it exists
+    if (paramTitle) {
+      setCrumbTitle(paramTitle);
+    }
+  }, [paramTitle]);
```
Comment (marcellmueller) [inline client/app/components/navigation/Bread.tsx:101]: I noticed is that the title takes a second or so to pop in. It seems to take less time to load if we do something like `const [crumbTitle, setCrumbTitle] = useState<string>(paramTitle ?? "");`

What is the purpose of the useEffect here? Does the param not exist sometimes until after the page is loaded?

File: client/app/components/form/SelectOperatorForm.tsx [hunk 2]
```diff
@@ -37,9 +37,20 @@ export default function SelectOperatorForm({
           setErrorList([{ message: response.error }]);
           return;
         }
-        // If the response is an array, we want the first element's id
-        let operatorId = response.length > 0 ? response[0].id : response.id;
-        push(`/dashboard/select-operator/confirm/${operatorId}`);
+        console.log(response);
```
Comment (marcellmueller) [inline client/app/components/form/SelectOperatorForm.tsx:40]: console.log :)

**Review by marcellmueller — COMMENTED** [review]
Nice work 🥇 This is working well and a huge improvement! Also will help us in the future if the designers want fancier breadcrumbs 🥳

File: client/app/components/navigation/Bread.tsx (hunk 1 above)
Comment (shon-button) [reply client/app/components/navigation/Bread.tsx:101]: The param will not exist when using page navigation buttons; so, setting up an effect to set the update crumbTitle whenever paramTitle changes and has a value...

**Note:**  changing to `const [crumbTitle, setCrumbTitle] = useState<string>(paramTitle ?? "");` doesn't seem to work on my end?

Comment (marcellmueller) [reply client/app/components/navigation/Bread.tsx:101]: Okay thanks for the explanation that makes sense!

**Review by marcellmueller — APPROVED** [review]
Thanks for the comments and the update 🙇 

I wonder if we can ask Hannah and the designers if it's okay to remove the ID column in the operator table in the future. They wanted a unique identifier in the breadcrumbs to let people would know where they were.

Now that we have this I'm not sure that column is useful.

**Pull request merged**